src/commands/deployment/correlate.ts

```diff
@@ -0,0 +1,182 @@
+import chalk from 'chalk'
+import {Command, Option} from 'clipanion'
+import simpleGit from 'simple-git'
+
+import {getCISpanTags} from '../../helpers/ci'
+import {gitRepositoryURL, gitLocalCommitShas, gitCurrentBranch} from '../../helpers/git/get-git-data'
+import {Logger, LogLevel} from '../../helpers/logger'
+import {retryRequest} from '../../helpers/retry'
+import {CI_PROVIDER_NAME, CI_ENV_VARS, GIT_REPOSITORY_URL, GIT_SHA} from '../../helpers/tags'
+import {getApiHostForSite, getRequestBuilder} from '../../helpers/utils'
+
+/**
+ * This command is a wrapper around the datadog-ci tag command, allowing customers to mark CI jobs
+ * as deployments and setting specific properties, like the environment or the revision in a simple way.
+ */
+export class DeploymentCorrelateCommand extends Command {
+  public static paths = [['deployment', 'correlate']]
+
+  public static usage = Command.Usage({
+    category: 'CI Visibility',
+    description: 'Correlate gitOps CD deployments with CI pipelines.',
+    details: `
+      This command will correlate the pipeline with a gitOps CD deployment.\n
+      It does not work for every setup, see README for details.
+    `,
+    examples: [['Correlate an argoCD deployment', 'datadog-ci deployment correlate --provider argocd']],
+  })
+
+  private cdProviderParam = Option.String('--provider')
+  private cdProvider!: string
```

Comment:
Do we really need this? Could we just use `cdProviderParam` after validating that is present?